### DISCUSSION TITLE
Using in threads from module

### DISCUSSION THREAD

**Discussion by stolx** [thesofproject/sof#7971] [General]
With some new modules we are designing there is a challenge of squeezing computation that is required by parameters change.
It will be beneficial to be able to create control thread within the module, to separate high-priority audio process task and lesser priority parameters updates. 
Is it possible to use threads from inside the module_adapter`s module?
And if not, maybe there is some approach you can suggest (other than computations optimization, which is already ongoing)?

**Comment by lgirdwood** [comment]
I think tday we are doing all non process() module work in the IPC thread and this is probably OK for current modules. However, there may be modules that require more cycles that would delay IPC (if using IPC thread) and will require some extension module API to schedule work. e.g. 
```c
int module_work_schedule_idle(module, work_func, arg);
int module_work_cancel(module, work_func, arg);
```
Which would call the Zephyr scheduling APIs with the appropriate priorities for the module type.

**Reply by marcinszkudlinski** [reply]
As long it is for sideband operations, not for audio stream processing, it should be fine to use a separate thread in module

Keep in mind that LL modules (current only module types) must finish the processing within 1 LL cycle
So I would not use any mutextes that may block stream processing, etc. 

upcoming DP module type will have its own dedicated thread (one per module instance) for data processing and flexible processing time
